src/Atmos/Model/precipitation.jl

```diff
@@ -63,115 +63,79 @@ Precipitation model with rain only.
+vars_state(::Rain, ::GradientFlux, FT) = @vars(∇q_rai::SVector{3, FT})
 
 function atmos_nodal_update_auxiliary_state!(
-    rain::Rain,
+    precip::Rain,
     atmos::AtmosModel,
     state::Vars,
     aux::Vars,
     t::Real,
-)
-    FT = eltype(state)
-    q_rain = state.precipitation.ρq_rain / state.ρ
-    if q_rain > FT(0) # TODO - need a way to prevent negative values
-        aux.precipitation.terminal_velocity = terminal_velocity(q_rain, state.ρ)
-    else
-        aux.precipitation.terminal_velocity = FT(0)
-    end
-
-    ρ = state.ρ
-    ρinv = 1 / state.ρ
-    p = aux.pressure
-    # TODO - ensure positive definite
-    q_tot = max(FT(0), state.ρq_tot * ρinv)
-    q_rai = max(FT(0), state.ρq_rain * ρinv)
-
-    # current state
-    ts = recover_thermo_state(rain, state, aux)
-    # q     = PhasePartition(q_tot, q_liq, q_ice)
-    q = PhasePartition(ts)
-    # T     = air_temperature(e_int, q)
-    T = air_temperature(ts)
-    # equilibrium state at current T
-    q_eq = PhasePartition_equil(ts)
-
-    # tendency from cloud water condensation/evaporation
-    # src_q_liq = conv_q_vap_to_q_liq(q_eq, q)# TODO - temporary handling ice
-
-    # tendencies from rain
-    src_q_rai_evap = conv_q_rai_to_q_vap(atmos.param_set, q_rai, q, T, p, ρ)
-    src_q_rai_acnv = conv_q_liq_to_q_rai_acnv(atmos.param_set, q.liq)
-    src_q_rai_accr = conv_q_liq_to_q_rai_accr(atmos.param_set, q.liq, q_rai, ρ)
-
-    aux.precipitation.src_q_rai_tot =
-        src_q_rai_acnv + src_q_rai_accr + src_q_rai_evap
-end
+) end
 
 function compute_gradient_argument!(
-    rain::Rain,
+    precip::Rain,
     transform::Vars,
     state::Vars,
     aux::Vars,
     t::Real,
 )
-    transform.precipitation.q_rain = state.precipitation.ρq_rain / state.ρ
+    ρinv = 1 / state.ρ
+    transform.precipitation.q_rai = state.precipitation.ρq_rai * ρinv
 end
 
 function compute_gradient_flux!(
-    rain::Rain,
+    precip::Rain,
     diffusive::Vars,
     ∇transform::Grad,
     state::Vars,
     aux::Vars,
     t::Real,
-    ρν::Union{Real, AbstractMatrix},
-    D_T,
 )
-    # diffusive flux
-    diffusive.precipitation.ρd_q_rain =
-        state.ρ .* (-D_T) .* ∇transform.precipitation.q_rain
+    # diffusive flux of q_tot
+    diffusive.precipitation.∇q_rai = ∇transform.precipitation.q_rai
 end
 
 function flux_precipitation!(
-    rain::Rain,
+    precip::Rain,
     atmos::AtmosModel,
     flux::Grad,
     state::Vars,
     aux::Vars,
     t::Real,
 )
+    FT = eltype(state)
     u = state.ρu / state.ρ
-    flux.precipitation.ρq_rain += state.precipitation.ρq_rain * u
+    q_rai = state.precipitation.ρq_rai / state.ρ
+
+    v_term::FT = FT(0)
+    if q_rai > FT(0)
+        v_term = terminal_velocity(
+            atmos.param_set,
+            atmos.param_set.microphys.rai,
+            state.ρ,
+            q_rai,
+        )
+    end
+
+    flux.precipitation.ρq_rai +=
+        state.precipitation.ρq_rai * (u - SVector(FT(0), FT(0), v_term))
```

Comment:
I think it would be better to define the terminal velocity as a downward pointing vertical vector from the outset, and then just add that vector velocity here. It seems to me that what is written here only works in a Cartesian box but not on a sphere (where the third vector component is not necessarily vertical).

Reply:
Replaced with: 

https://github.com/CliMA/ClimateMachine.jl/blob/47a3c983568896118ba6afba7c9b26f3d9ad2329/src/Atmos/Model/precipitation.jl#L123-L124